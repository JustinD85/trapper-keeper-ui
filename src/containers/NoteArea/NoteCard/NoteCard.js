import React from "react";
import { NoteItems } from "../../";
class NoteCard extends React.Component {
  constructor({ note }) {
    super(note);
    this.state = {
      title: note.title,
      noteItems: note.noteItems
    };
  }

  render() {
    const { title, noteItems } = this.props.note;
    return (
      <div>
        <h2>{title}</h2>
        <ul>
          <NoteItems noteItems={noteItems} />
        </ul>
      </div>
    );
  }
}

export default NoteCard;
