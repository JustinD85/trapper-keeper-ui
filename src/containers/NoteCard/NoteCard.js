import React from "react";
import { NoteItems } from "../";
import PropTypes from "prop-types";
import { Link, NavLink } from "react-router-dom";
import { deleteNote } from '../../thunks';
import { connect } from 'react-redux';
import { Card, IconButton, CardHeader, CardContent, Tooltip } from '@material-ui/core';
import { DeleteOutline } from '@material-ui/icons'
import { withStyles } from '@material-ui/core/styles'

const styles = {
  link: {
    textDecoration: 'none',
  },
  card: {
    margin: 10,
  },
  title: {
    color: '#48494a',
  },
  delete: {
    float: 'right',
    margin: '3px'
  }
}

export const NoteCard = ({ note, deleteNote, classes }) => (
  <Tooltip title='Click to edit!' placement='top' enterDelay={2000}>
    <Card className={classes.card}>
      <Link to='/' className={classes.delete}>
        <Tooltip title='Delete this note!'>
          <IconButton onClick={() => deleteNote(note)}>
            <DeleteOutline />
          </IconButton>
        </Tooltip>
      </Link>
      <Link to={`/notes/${note.id}`} className={classes.link}>
        <CardHeader
          title={note.title}
          classes={{ title: classes.title }}
        />
        <CardContent>
          <ul>
            <NoteItems noteItems={note.noteItems} />
          </ul>
        </CardContent>
      </Link>
    </Card>
  </Tooltip>
);

export const mapDispatchToProps = (dispatch) => ({
  deleteNote: (note) => dispatch(deleteNote(note))
});

export default withStyles(styles)(connect(null, mapDispatchToProps)(NoteCard));

NoteCard.propTypes = {
  note: PropTypes.object,
  deleteNote: PropTypes.func
};
