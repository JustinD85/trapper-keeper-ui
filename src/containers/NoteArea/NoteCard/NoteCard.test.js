import React from 'react';
import { shallow } from 'enzyme';
import { NoteCard } from './NoteCard';

describe('NoteCard', () => {
  let wrapper;
  const mockNote = {id: 1, title: 'Best Note', noteItems: [{text: 'lala'}, {text: 'hooray'}]}
  beforeEach(() => {
    wrapper = shallow(<NoteCard note={mockNote}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})