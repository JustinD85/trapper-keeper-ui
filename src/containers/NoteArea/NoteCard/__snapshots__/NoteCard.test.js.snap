// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCard should match the snapshot 1`] = `
<Link
  replace={false}
  to="/notes/1"
>
  <h2>
    Best Note
  </h2>
  <ul>
    <NoteItems
      noteItems={
        Array [
          Object {
            "text": "lala",
          },
          Object {
            "text": "hooray",
          },
        ]
      }
    />
  </ul>
</Link>
`;
